src/app/components/itemlist.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useEffect, useState } from "react";
+import useFetchAdvocates, { Advocate } from "../hooks/useFetchAdvocates";
+
+interface ItemListProps {
+	searchText: string | null;
+}
+
+const ItemList: React.FC<ItemListProps> = ({ searchText }: ItemListProps) => {
+
+	const [currentPage, setCurrentPage] = useState(1);
+	const [pageSize,] = useState(10);
+	const offset = (currentPage - 1) * pageSize;
+
+	const { advocates, resultCount, loading } = useFetchAdvocates("/api/advocates", pageSize, offset);
+
+	const totalPages = Math.ceil((resultCount ?? 0) / pageSize);
+
+	const nextPage = () => {
+		if (currentPage < totalPages) {
+			setCurrentPage(currentPage + 1);
+		}
+	};
+
+	const prevPage = () => {
+		if (currentPage > 1) {
+			setCurrentPage(currentPage - 1);
+		}
+	};
+
+	const filteredItems = advocates && advocates.filter(
```

Comment:
This is great for doing searches on the UI only. But if we were to expand this a bit further. Maybe have to look at server side searching.